### DISCUSSION TITLE
[@types/webpack-dev-server] package.json directive is incompatible with webpack@5

### DISCUSSION THREAD

**Discussion by ermik** [DefinitelyTyped/DefinitelyTyped#55865] [Issues with a @types package]
Latest version of `@types/webpack-dev-server` distributed via NPM (`4.1.0`) has a `package.json` dependency on `"@types/webpack": "^4"` which will cause the Webpack config type-checking to fail. As far as I can see the source code for these declarations has no explicit dependency on `@types/webpack` so I have no clue how this is happening...

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
> Latest version of `@types/webpack-dev-server` distributed via NPM (`4.1.0`) has a `package.json` dependency on `"@types/webpack": "^4"` which will cause the Webpack config type-checking to fail. As far as I can see the source code for these declarations has no explicit dependency on `@types/webpack` so I have no clue how this is happening...

For now `webpack-dev-server` supports both v5 and v4:
https://github.com/webpack/webpack-dev-server/blob/master/migration-v4.md#-breaking-changes (version support)
https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L1080
there is no clean way to implement this on DT. This is the reason AFAIK
